auth/cache.go

```diff
@@ -0,0 +1,46 @@
+/*
+Copyright 2023 The Flux authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package auth
+
+import (
+	"sync"
+	"time"
+)
+
+var once sync.Once
+var cache Store
+
+// InitCache intializes the pacakge cache with the provided cache object.
+// Consumers that want automatic caching when using `GetRegistryAuthenticator()`
+// or `GetGitCredentials()` must call this before. It should only be called once,
+// all subsequent calls will be a no-op.
+func InitCache(s Store) {
```

Comment:
I think it would be good if this returned an error for subsequent calls, to indicate it was already set.